src/modules/reports/ReportSubmissionForm.tsx

```
@@ -206,7 +205,22 @@ const ReportSubmissionForm = () => {
+  
+  const handleDateFieldOpen = useCallback((fieldName: string) => {
+    setOpenPickers((prev) => ({ ...prev, [fieldName]: true }));
+  }, []);
+
+  const handleDateFieldClose = useCallback((fieldName: string) => {
+    setOpenPickers((prev) => ({ ...prev, [fieldName]: false }));
+  }, []);
+
+  const handleDateFieldChange = useCallback(
+    (fieldName: string) => (date: Date | null) => {
+      handleChange(fieldName, date ? format(date, 'yyyy-MM-dd') : '');
```

ESLint

[error] Error: Cannot access variable before it is declared

`handleChange` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/src/modules/reports/ReportSubmissionForm.tsx:220:7
  218 |   const handleDateFieldChange = useCallback(
  219 |     (fieldName: string) => (date: Date | null) => {
> 220 |       handleChange(fieldName, date ? format(date, 'yyyy-MM-dd') : '');
      |       ^^^^^^^^^^^^ `handleChange` accessed before it is declared
  221 |     },
  222 |     [],
  223 |   );

/home/jailuser/git/src/modules/reports/ReportSubmissionForm.tsx:300:3
  298 |     field.name === 'serviceLocationName';
  299 |
> 300 |   const handleChange = (name: string, value: $TsFixMe) => {
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 301 |     setFormData((prev) => ({ ...prev, [name]: value }));
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 302 |     setValidationErrors((prev) => {
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 303 |       const next = { ...prev };
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 304 |       delete next[name];
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 305 |       return next;
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 306 |     });
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 307 |   };
      | ^^^^^ `handleChange` is declared here
  308 |
  309 |       const handleSmallGroupChange = (group: DynamicGroup | null) => {
  310 |     setFormData((prev) => {

(react-hooks/immutability)